Build error after wlroots!4482 "presentation-time: drop wlr_presentation arg"

### Please fill out the following:
- **Sway Version:**
sway version 1.9-dev-22d0dd8b (Dec 27 2023, branch 'master')

- **Debug Log:**

Compiling with Clang
```
INFO: autodetecting backend as ninja                                                                                                                            
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/makepkg/sway-git/src/build
ninja: Entering directory `/tmp/makepkg/sway-git/src/build'
[68/312] Compiling C object sway/sway.p/desktop_output.c.o
FAILED: sway/sway.p/desktop_output.c.o 
clang -Isway/sway.p -Isway -I../sway-git/sway -Iinclude -I../sway-git/include -Iprotocols -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -flto -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wno-format-zero-length -Wundef -Wvla '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.9-dev-22d0dd8b (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../sway-git/= -march=native -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -Wno-incompatible-pointer-types -flto=auto -fPIE -pthread -MD -MQ sway/sway.p/desktop_output.c.o -MF sway/sway.p/desktop_output.c.o.d -o sway/sway.p/desktop_output.c.o -c ../sway-git/sway/desktop/output.c
../sway-git/sway/desktop/output.c:535:3: error: too many arguments to function call, expected 2, have 3
                wlr_output);
                ^~~~~~~~~~
/usr/include/wlr/types/wlr_presentation_time.h:103:6: note: 'wlr_presentation_surface_scanned_out_on_output' declared here
void wlr_presentation_surface_scanned_out_on_output(struct wlr_surface *surface,
     ^
1 error generated.
[78/312] Compiling C object sway/sway.p/desktop_render.c.o
FAILED: sway/sway.p/desktop_render.c.o 
clang -Isway/sway.p -Isway -I../sway-git/sway -Iinclude -I../sway-git/include -Iprotocols -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -flto -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wno-format-zero-length -Wundef -Wvla '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.9-dev-22d0dd8b (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../sway-git/= -march=native -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -Wno-incompatible-pointer-types -flto=auto -fPIE -pthread -MD -MQ sway/sway.p/desktop_render.c.o -MF sway/sway.p/desktop_render.c.o.d -o sway/sway.p/desktop_render.c.o -c ../sway-git/sway/desktop/render.c
../sway-git/sway/desktop/render.c:155:3: error: too many arguments to function call, expected 2, have 3
                wlr_output);
                ^~~~~~~~~~
/usr/include/wlr/types/wlr_presentation_time.h:95:6: note: 'wlr_presentation_surface_textured_on_output' declared here
void wlr_presentation_surface_textured_on_output(struct wlr_surface *surface,
     ^
1 error generated.
[85/312] Compiling C object sway/sway.p/input_cursor.c.o
../sway-git/sway/input/cursor.c:930:36: warning: unused variable 'constraint' [-Wunused-variable]
        struct wlr_pointer_constraint_v1 *constraint = cursor->active_constraint;
                                          ^
1 warning generated.
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'sway-git-r7230.bbabb9aa-1': 
error: packages failed to build: sway-git-r7230.bbabb9aa-1
```

Compiling with GCC  
```
INFO: autodetecting backend as ninja                                                                                    
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/makepkg/sway-git/src/build
ninja: Entering directory `/tmp/makepkg/sway-git/src/build'
[65/312] Compiling C object sway/sway.p/desktop_output.c.o
FAILED: sway/sway.p/desktop_output.c.o 
cc -Isway/sway.p -Isway -I../sway-git/sway -Iinclude -I../sway-git/include -Iprotocols -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -flto=auto -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wno-format-zero-length -Wundef -Wvla '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.9-dev-22d0dd8b (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../sway-git/= -march=native -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -flto=auto -fPIE -pthread -MD -MQ sway/sway.p/desktop_output.c.o -MF sway/sway.p/desktop_output.c.o.d -o sway/sway.p/desktop_output.c.o -c ../sway-git/sway/desktop/output.c
../sway-git/sway/desktop/output.c: In function ‘scan_out_fullscreen_view’:
../sway-git/sway/desktop/output.c:534:62: warning: passing argument 1 of ‘wlr_presentation_surface_scanned_out_on_output’ from incompatible pointer type [-Wincompatible-pointer-types]
  534 |         wlr_presentation_surface_scanned_out_on_output(server.presentation, surface,
      |                                                        ~~~~~~^~~~~~~~~~~~~
      |                                                              |
      |                                                              struct wlr_presentation *
In file included from ../sway-git/sway/desktop/output.c:16:
/usr/include/wlr/types/wlr_presentation_time.h:103:73: note: expected ‘struct wlr_surface *’ but argument is of type ‘struct wlr_presentation *’
  103 | void wlr_presentation_surface_scanned_out_on_output(struct wlr_surface *surface,
      |                                                     ~~~~~~~~~~~~~~~~~~~~^~~~~~~
../sway-git/sway/desktop/output.c:534:77: warning: passing argument 2 of ‘wlr_presentation_surface_scanned_out_on_output’ from incompatible pointer type [-Wincompatible-pointer-types]
  534 |         wlr_presentation_surface_scanned_out_on_output(server.presentation, surface,
      |                                                                             ^~~~~~~
      |                                                                             |
      |                                                                             struct wlr_surface *
/usr/include/wlr/types/wlr_presentation_time.h:104:28: note: expected ‘struct wlr_output *’ but argument is of type ‘struct wlr_surface *’
  104 |         struct wlr_output *output);
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~
../sway-git/sway/desktop/output.c:534:9: error: too many arguments to function ‘wlr_presentation_surface_scanned_out_on_output’
  534 |         wlr_presentation_surface_scanned_out_on_output(server.presentation, surface,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/wlr/types/wlr_presentation_time.h:103:6: note: declared here
  103 | void wlr_presentation_surface_scanned_out_on_output(struct wlr_surface *surface,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[78/312] Compiling C object sway/sway.p/desktop_render.c.o
FAILED: sway/sway.p/desktop_render.c.o 
cc -Isway/sway.p -Isway -I../sway-git/sway -Iinclude -I../sway-git/include -Iprotocols -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -flto=auto -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wno-format-zero-length -Wundef -Wvla '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.9-dev-22d0dd8b (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../sway-git/= -march=native -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -flto=auto -fPIE -pthread -MD -MQ sway/sway.p/desktop_render.c.o -MF sway/sway.p/desktop_render.c.o.d -o sway/sway.p/desktop_render.c.o -c ../sway-git/sway/desktop/render.c
../sway-git/sway/desktop/render.c: In function ‘render_surface_iterator’:
../sway-git/sway/desktop/render.c:154:59: warning: passing argument 1 of ‘wlr_presentation_surface_textured_on_output’ from incompatible pointer type [-Wincompatible-pointer-types]
  154 |         wlr_presentation_surface_textured_on_output(server.presentation, surface,
      |                                                     ~~~~~~^~~~~~~~~~~~~
      |                                                           |
      |                                                           struct wlr_presentation *
In file included from ../sway-git/include/sway/server.h:16,
                 from ../sway-git/include/sway/input/input-manager.h:7,
                 from ../sway-git/sway/desktop/render.c:20:
/usr/include/wlr/types/wlr_presentation_time.h:95:70: note: expected ‘struct wlr_surface *’ but argument is of type ‘struct wlr_presentation *’
   95 | void wlr_presentation_surface_textured_on_output(struct wlr_surface *surface,
      |                                                  ~~~~~~~~~~~~~~~~~~~~^~~~~~~
../sway-git/sway/desktop/render.c:154:74: warning: passing argument 2 of ‘wlr_presentation_surface_textured_on_output’ from incompatible pointer type [-Wincompatible-pointer-types]
  154 |         wlr_presentation_surface_textured_on_output(server.presentation, surface,
      |                                                                          ^~~~~~~
      |                                                                          |
      |                                                                          struct wlr_surface *
/usr/include/wlr/types/wlr_presentation_time.h:96:28: note: expected ‘struct wlr_output *’ but argument is of type ‘struct wlr_surface *’
   96 |         struct wlr_output *output);
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~
../sway-git/sway/desktop/render.c:154:9: error: too many arguments to function ‘wlr_presentation_surface_textured_on_output’
  154 |         wlr_presentation_surface_textured_on_output(server.presentation, surface,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/wlr/types/wlr_presentation_time.h:95:6: note: declared here
   95 | void wlr_presentation_surface_textured_on_output(struct wlr_surface *surface,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[82/312] Compiling C object sway/sway.p/desktop_launcher.c.o
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'sway-git-r7230.bbabb9aa-1': 
error: packages failed to build: sway-git-r7230.bbabb9aa-1
```

- **Configuration File:**
 - N/A

- **Stack Trace:**
 - N/A

- **Description:**
  
`wlr_presentation` arg has been dropped in multiple function. 
Need to chase wlroots's change. [wlroots!4482](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4482)

--- a/sway/desktop/output.c
+++ b/sway/desktop/output.c
@@ -531,7 +531,7 @@ static bool scan_out_fullscreen_view(struct sway_output *output,
 		return false;
 	}
 
-	wlr_presentation_surface_scanned_out_on_output(server.presentation, surface,
+	wlr_presentation_surface_scanned_out_on_output(surface,
 		wlr_output);
 
 	return wlr_output_commit_state(wlr_output, pending);

--- a/sway/desktop/render.c
+++ b/sway/desktop/render.c
@@ -151,7 +151,7 @@ static void render_surface_iterator(struct sway_output *output,
 	render_texture(data->ctx, texture,
 		&src_box, &dst_box, &clip_box, surface->current.transform, alpha);
 
-	wlr_presentation_surface_textured_on_output(server.presentation, surface,
+	wlr_presentation_surface_textured_on_output(surface,
 		wlr_output);
 }
 

